src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Validation.cs

```diff
@@ -154,10 +164,10 @@ internal static unsafe partial class Utf8Utility
 
                         continue; // need to perform a bounds check because we might be running out of data
 
-                    Sse2LoopTerminatedEarlyDueToNonAsciiData:
+                    LoopTerminatedEarlyDueToNonAsciiData:
 
                         Debug.Assert(BitConverter.IsLittleEndian);
```

Comment:
Unlike x86, ARM64 can be big endian and while I don't believe CoreCLR currently supports any big endian output, other platforms like Mono do.

Reply:
@BruceForstall, @jeffhandley. Is supporting big endian something we want to target here for .NET 5 or should we leave it as one of the things that will need to be updated when Mono adds support for the arm intrinsics?

Reply:
We can always change the check to `AdvSimd.IsSupported && BitConverter.IsLittleEndian`, which will still be a JIT constant and which will cause big endian machines to use the software fallback.

Reply:
I don't understand the scenarios where big endian is required, but your suggestion to guard using IsLittleEndian sounds reasonable.

Reply:
I went through the same thought process a day or so back. The only thing I'd add is that it's also not straightforward to test a big endian environment at the moment. So we'd only have code-review guarantees that the code is big endian correct.

Reply:
I added little endian checks in my last commit. Please take a look whenever you get a chance.